### ISSUE TITLE
[Information] Scraping with 'thegamesdb' - No results on games with special chars in name

### ISSUE THREAD

**Issue by GM-Script-Writer-62850** [Aloshi/EmulationStation#259]
These games exist in the database but are not found when scrapping
[Pac-Man](https://thegamesdb.net/search/?string=Pac-Man&function=Search&stringPlatform=22)
[Jr. Pac-Man](https://thegamesdb.net/search/?string=Jr.%20Pac-Man&function=Search&stringPlatform=22)
[Ms. Pac-Man](https://thegamesdb.net/search/?string=Ms.%20Pac-Man&function=Search&stringPlatform=22)
yes the rom files are names appropriately

```
~$ ls atari2600/ | grep -i pac | grep -i man
Jr. Pac-Man.A26
Ms. Pac-Man.A26
Pac-Man.A26

```

**Comment by flexage** [comment]
@nilsbyte Hey, I'm the lead dev over at thegamesdb.net (when I actually get time to work on it).

A little over a year ago, I set up an ElasticSearch server for TGDB, in an attempt to get rid of some of the many issues with the existing MySQL-based searching, but I never actually got around to updating the API to use it.

I've put in a little time and moved http://thegamesdb.net/api/GetGamesList.php over to use the ElasticSearch server (as per commit https://github.com/TheGamesDB/TheGamesDB/commit/b120524f19bf01a27c5fcf29ea46615bf40e5b30), just this one API call first as a test.

In limited tests on my dev server, there were many benefits over the MySQL based implementation, including:

   > Titles with special characters, can now be searched properly: you should either include the special character in your name parameter (pac-man), or replace it with a space(pac man)

   > Search results are ordered in a more relevant manner (based on their closeness to key tokens from the search string), however there are some limitations: searching "halo" when looking for the original halo will return "halo 2" as the first result, this is actually the desired behavior for that string as the games proper title is "halo: combat evolved" and should be searched as such.
   
   > Game titles with numbers return more appropriately ordered results (searching "far cry 2" will return an exact match first, followed by "far cry")

   > Searches for game titles with roman numerals ("final fantasy vii") will be matched even if you use an integer in place of the roman numerals

   > Numerous additional small enhancements, including speed

It already seems a lot better from what I can tell, but if you could feed back some information from your projects use of this API call it would be much appreciated, your project makes much more practical use of game searching than any of my local tests could.

I hope it goes some way to resolving the issues that yourselves and your users have been experiencing with scraping games from our site, pending feedback from yourselves I will also be happy to move the GetGame.php api call over to use the new search engine.

**Comment by Enverex** [comment]
I'll run my scraper/indexer against it tomorrow (after clearing the local TGDB cache). There's around 32000 games it has to look up (but it won't need to do all of them as some are already completed/fully populated). I'll let you know if it picks up a significant amount of new results.

**Comment by flexage** [comment]
@Enverex Cool, thanks for that. I'll be interested to know of the results :wink:

**Comment by Enverex** [comment]
I've just checked, my script uses "http://thegamesdb.net/api/GetGame.php" rather than GetGamesList.php so I don't think I can do anything with this yet. I'll double check what it's doing and if I can switch.

**Comment by Enverex** [comment]
Also "alias" names for Platforms no-longer seem to work.

http://thegamesdb.net/api/GetPlatformsList.php specifies...

    <Platform>
    <id>4914</id>
    <name>Amstrad CPC</name>
    <alias>amstrad-cpc</alias>
    </Platform>

But trying to use "amstrad-cpc" (which used to work) now returns "The specified platform was not valid".

**Comment by flexage** [comment]
You are right, I'll look into that issue over the weekend, and see if I can tweak the query a bit to improve the scorings. Thanks for reporting this issue.

**Comment by Enverex** [comment]
Ran into this recently:

	<?xml version="1.0" encoding="UTF-8" ?>
	Could not connect: Can't create a new thread (errno 11); if you are not out of available memory, you can consult the manual for a possible OS-dependent bug

... as the text returned for a query. The URL, in case it's relevant, was "http://thegamesdb.net/api/GetGamesList.php?name=Witchfiend&platform=Sinclair%20ZX%20Spectrum".

**Comment by MarcDK** [comment]
I got still the problem that "F-Zero" can not be scrapped. Even if I rename the rom to fzero or f zero the result is always empty.

**Comment by Schroedingers-Cat** [comment]
So what is the status of this one?

**Comment by Enverex** [comment]
The official EmulationStation is dead, see https://github.com/RetroPie/EmulationStation instead.

**Comment by telmofcosta** [comment]
Just realised that "F-Zero" can be scrapped if the special character is escaped with a backslash. So, instead of searching by "F-Zero", change the input text to "F\-Zero" and it will work.

**Comment by MEPoss** [comment]
@telmofcosta did you try with "F\-Zero"? Did it work?

**Comment by telmofcosta** [comment]
@MEPoss I don't quite get your question. I used scraper on my SNES collection. All games worked except for F-Zero. I then changed the input text manually to "F\\-Zero", searched with it and got a  match.

**Comment by MEPoss** [comment]
@telmofcosta ok, I just wanted to know if it worked or if you were just guessing

**Comment by MEPoss** [comment]
@telmofcosta I tried with both `F\-Zero` and `fzero` .zip but it does not work.

**Comment by mtoensing** [comment]
fzero in one word worked for me.

**Comment by MEPoss** [comment]
@mtoensing with the built-in scraper of EmulationStation?

**Comment by Dimensional** [comment]
Interestingly I did a search using the API directly for F-Zero using the exactname argument, and it displayed the title. Using just the name argument didn't yield any results when I searched by that title, but fzero worked.

**Comment by r4ndom4is** [comment]
try Spy vs. Spy as in http://thegamesdb.net/game/6564/

i'm unable to find with any combination
